File: get-statistics/get-statistics-with-Vega-API-client.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (gRPC)

Apps/Libraries:
- gRPC: Vega-API-client (https://pypi.org/project/Vega-API-client/)
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import os

# __import_client:
import vegaapiclient as vac
# :import_client__

node_url_grpc = os.getenv("NODE_URL_GRPC")

# __get_statistics:
# Request the statistics for a node on Vega
data_client = vac.VegaTradingDataClient(node_url_grpc)
response = data_client.Statistics(vac.api.trading.StatisticsRequest())
print("Statistics:\n{}".format(response))
# :get_statistics__

File: get-statistics/get-statistics-with-Vega-API-client.go
package main

import (
	"context"
	"fmt"
	"os"

	api "code.vegaprotocol.io/protos/vega/api/v1"
	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := os.Getenv("NODE_URL_GRPC")
	if len(nodeURLGrpc) == 0 {
		panic("NODE_URL_GRPC is null or empty")
	}

	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewCoreServiceClient(conn)
	// __get_statistics:
	// Request the statistics for a node on Vega
	request := api.StatisticsRequest{}
	statistics, err := dataClient.Statistics(context.Background(), &request)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Network statistics: %s", statistics)
	// :get_statistics__
}
